calico_upgrade/pkg/migrate/felixconfig.go

```diff
@@ -187,6 +189,27 @@ func (m *migrationHelper) parseFelixConfigV1IntoResourceV3(
 			configStrValue = convertLogLevel(configStrValue)
 		}
 
+		_, ok = fieldValue.Interface().(*metav1.Duration); if ok {
+			if duration, err := strconv.ParseFloat(configStrValue, 64); err != nil {
+				logCxt.WithError(err).Info("Failed to parse float for Duration field")
+				data.ConversionErrors = append(data.ConversionErrors, ConversionError{
+					Cause:   fmt.Errorf("failed to parse float for Duration field: %v", err),
+					KeyV1:   keysv1[configName],
+					ValueV1: configStrValue,
+					KeyV3:   resourceToKey(res),
+			    })
+			} else {
+				switch field.Tag.Get("configv1timescale") {
+				case "milliseconds":
+					fieldValue.Set(reflect.ValueOf(&metav1.Duration{Duration: time.Duration(duration) * time.Millisecond}))
```

Comment:
Not sure how this is working.    You are doing the following:

`time.Duration(<float>) * time.Millisecond`

I would have though that time.Duration(<float>) would end up rounding up the float.  Does it not?

Reply:
IIUC it would need to be:

`time.Duration(duration * float64(time.Millisecond))`

or am i talking rubbish?

Reply:
yeah, because you are multiplying by time.Second and time.Millisecond for the s and ms fields respectively, these will never have anything after the decimal point when converted to s and ms

Reply:
yea spot on, fixing up the tests/code now